### DISCUSSION TITLE
Automatic removal of abandoned shopping carts

### DISCUSSION THREAD

**Discussion by eveneme** [PrestaShop/PrestaShop#41067] [Ideas]
#### Is your feature request related to a problem?

No

#### Describe the solution you'd like

I would like abandoned shopping baskets to be automatically deleted if the payment has not been made, so that it does not interfere with stock! 

#### Alternatives you've considered

The module is built into prestashop 

#### Additional context

If the stock is small, it is a very troublesome topic because this product is suspended in abandoned baskets and you need to remove it to reappear in stock!

**Comment by khouloudbelguith** [comment]
Hi @eveneme,

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thanks!

**Comment by Hlavtox** [comment]
Hi, shopping cart does not reserve the stock for anybody, it does not interfere with stock. Only you as a customer see the product quantity reduced by the amount of items in your cart.

And if you are talking about created unpaid orders, its very easy to make your own script.
- Select orders from ps_orders where current_status is unpaid and time difference from order creation is bigger than 1 day.
- Cancel each order you find. Or send a notification to customer.

**Comment by DracoBluee** [comment]
it is about unpaid orders
